tfjs-core/src/ops/segment_ops.ts

```diff
@@ -95,67 +94,17 @@ function gather_<T extends Tensor>(
   axis = parseAxisParam(axis, $x.shape)[0];
   const shapeInfo = collectGatherOpShapeInfo($x, $indices, axis);
 
-  const grad = (dy: T, saved: Tensor[]) => {
-    const [$indices] = saved;
-    const derX = () => {
-      const paramsShape = $x.shape;
-      const indicesSize = $indices.size;
-
-      const outerShape = paramsShape.slice(0, axis);
-      const outerDims = outerShape.length;
-      const innerShape = paramsShape.slice(axis, paramsShape.length).slice(1);
-      const innerDims = innerShape.length;
-
-      const outerAxesIndices = arrayRange(0, outerDims);
-      const innerAxesIndices =
-          arrayRange(outerDims + 1, outerDims + 1 + innerDims);
-
-      const valuesShape = arrayConcat([outerShape, [indicesSize], innerShape]);
-
-      const values = dy.reshape(valuesShape);
-      const reshapedIndices = $indices.reshape([indicesSize]);
-
-      const transposeDims =
-          arrayConcat([[outerDims], outerAxesIndices, innerAxesIndices]);
-      const valuesTranspose = values.transpose(transposeDims);
-      let paramsGrad = unsortedSegmentSum(
-          valuesTranspose, reshapedIndices as Tensor1D, $x.shape[axis]);
-
-      const invertTransposeDims = getUndoAxesPermutation(transposeDims);
-      paramsGrad = paramsGrad.transpose(invertTransposeDims);
-
-      return paramsGrad as T;
-    };
-    return {x: derX, indices: () => $indices};
-  };
+  const attrs = {axis};
   return (ENGINE.runKernelFunc(
               (backend, save) => {
                 const res = backend.gather($x, $indices.flatten(), axis);
```

Comment:
Could you also create a separate tfjs-core/src/ops/gather.ts file and move the gather definition there?